packages/server/lib/controllers/sync.controller.ts

```diff
@@ -228,13 +241,70 @@ class SyncController {
         }
     }
 
+    public async actionOrModel(req: Request, res: Response, next: NextFunction) {
+        try {
+            const environmentId = getEnvironmentId(res);
+            const providerConfigKey = req.get('Provider-Config-Key') as string;
+            const connectionId = req.get('Connection-Id') as string;
+            const action_or_model = req.params['action_or_model'] as string;
+            const category = req.params['category'] as string;
+
+            //TODO: add support for category per sync
+            if (category != "all") {
+                res.status(404).send({ message: `Unknown category ${category}` });
+                return;
+            }
+
+            if (!action_or_model) {
+                res.status(400).send({ message: 'Missing action or model name' });
+                return;
+            }
+
+            if (!connectionId) {
+                res.status(400).send({ message: 'Missing connection id' });
+
+                return;
+            }
+
+            if (!providerConfigKey) {
+                res.status(400).send({ message: 'Missing provider config key' });
+
+                return;
+            }
+
+            const {
+                success,
+                error,
+                response: connection
+            } = await connectionService.getConnection(connectionId as string, providerConfigKey as string, environmentId);
+
+            if (!success) {
+                errorManager.errResFromNangoErr(res, error);
+                return;
+            }
```

Comment:
There is a bit of duplication with the request validation. We can refactor it when we get rid of the old `sync/records` and `trigger/action` routes.